WordPressKit/WordPressComRestApi.swift

```diff
@@ -528,10 +539,13 @@ extension WordPressComRestApi {
 
         let mappedError = errorsMap[errorCode] ?? .unknown
         if mappedError == .invalidToken {
-            invalidTokenHandler?()
+            // Call `invalidTokenHandler in the main thread since it's typically used by the apps to present an authentication UI.
```

Comment:
Oof – good to preserve existing behaviour, but sad to add this dispatch here.

Do the apps care about the internal state of this object? Just wondering if we should use `DispatchQueue.main.sync` to wait for whatever handler to run before we carry on?

Reply:
When using Alamofire, this `processError` function is called from the main thread. That means, upon receiving an error, the `invalidTokenHandler` from the app will be called from the main thread, and later the specific app feature that sends the API request will receive a failure result from the main thread. And these two calls are probably called in the same runloop.

When using URLSession, this `processError` function is called from a background thread. That means, the `invalidTokenHandler` will be called asynchronously from the main thread, whilst the `processError` continues to run in the background which eventually calls the specific app feature code's failure block from the main thread. In this scenario, these two calls are definitely not called in the same runloop. But the `invalidTokenHandler` _should_ (considering it's dispatched to the main queue first) be called from the app feature code.

So, I think we don't need to wait for `invalidTokenHandler` to be called here.

Reply:
If it matches the existing behaviour, then fair enough – it still feels a little fragile IMHO(?) but that's for another day